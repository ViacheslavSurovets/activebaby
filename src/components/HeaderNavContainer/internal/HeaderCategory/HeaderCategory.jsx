import React, { useMemo } from 'react';
import { DropdownGroup, DropdownLink, ListItem } from '@components/Header/styles';
import { useTranslation } from 'react-i18next';
import Proptypes from 'prop-types';

const HeaderCategory = ( { headerCategoryData } ) => {
  const { t, i18n } = useTranslation ();


  const headerCategory = useMemo ( () => (
    headerCategoryData.map ( item => {
      return {
        key: item.id,
        listItems: item.listItems.map ( ( listItem, idx ) => {
          return {
            key: idx,
            text: t ( listItem.text ),
            path: listItem.path
          };
        } )
      };
    } )

  ), [ i18n.language ] );

  return (
    <>
      { headerCategory.map ( item =>
        <DropdownGroup key={ item.key }>
          { item.listItems.map ( ( listItem, idx ) =>
            <ListItem key={ (idx) }><DropdownLink
              to={ listItem.path }>{ listItem.text }</DropdownLink></ListItem>
          ) }
        </DropdownGroup>
      ) }
    </>
  );
};


export default HeaderCategory;

HeaderCategory.propTypes = {
  headerCategoryData: Proptypes.array.isRequired
};
