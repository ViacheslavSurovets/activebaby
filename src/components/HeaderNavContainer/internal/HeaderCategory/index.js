export { default as HeaderCategory } from './HeaderCategory';
