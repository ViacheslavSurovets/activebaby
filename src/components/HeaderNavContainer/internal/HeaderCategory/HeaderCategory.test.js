import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import HeaderCategory from './HeaderCategory';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('HeaderCategory', ()=> {
  it('HeaderCategory Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <HeaderCategory  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
