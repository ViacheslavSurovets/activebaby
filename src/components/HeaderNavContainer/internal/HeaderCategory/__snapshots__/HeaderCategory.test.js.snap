// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCategory HeaderCategory Snapshot 1`] = `ShallowWrapper {}`;
